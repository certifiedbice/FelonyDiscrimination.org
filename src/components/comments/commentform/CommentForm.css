#organization-comment-form-container {
    text-align: center;
}

#comment-form legend {
    width: 100%;
}

#comment-form legend h3 {
    text-align: center;
}

.comment-form-element-container {
    margin: 0 0 10px 0;
    text-align: center;
}

.comment-form-element-container label {
    display: block;
    margin: 0 0 5px 20px;
    text-align: left;
    font-size: 12px;
}

.comment-form-element-container input {
    width: 100%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    border-radius: 50px;
    padding: 5px;
    font-size: 12px;
    color: #e6e7e8;
}

.comment-form-element-container textarea {
    width: 92%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    border-radius: 25px;
    padding: 10px;
    color: #e6e7e8;
}

#comment-form-submit {
    width: 90%;
    background-color: #4378bc;
    border: 1px solid #e6e7e8;
    border-radius: 50px;
    padding: 5px;
    font-size: 12px;
    color: #e6e7e8;
    cursor: pointer;
}

#comment-form-submit:hover {
    width: 90%;
    background-color: #323232;
    border: 1px solid #e6e7e8;
    font-size: 12px;
    color: #e6e7e8;
}

/* Desktop CSS */
@media(min-width:991px){
    
    .comment-form-element-container textarea {
        width: 98%;
    }

    #comment-form-submit {
        width: 25%;
    }

    #comment-form-submit:hover {
        width: 25%;
    }

 }