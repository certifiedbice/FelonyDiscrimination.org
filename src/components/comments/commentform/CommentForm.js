import React,{Component} from 'react';
import {Link} from 'react-router-dom';
import CommentsApiService from '../../../services/comments-api-service';
import TokenService from '../../../services/token-service';
import './CommentForm.css';

export default class CommentForm extends Component{
    state={error:null};
    renderCommentForm(){
        return(
            <form id="comment-form" name="comment-form" aria-label="Comment form" onSubmit={this.handleSubmit}>
                <fieldset>
                    <legend><h3>Submit a Comment</h3></legend>
                        <div className="comment-form-element-container">
                            <label>Title:</label>
                            <input className="comment-form-element" name="title" id="comment-form-input" type="text" required aria-labelledby="Comment form email" placeholder="Search"/>
                        </div>
                        <div className="comment-form-element-container">
                            <label>Message:</label>
                            <textarea className="comment-form-element" name="comment" id="comment-form-message" rows="6" cols="27" required aria-labelledby="review-form-message"/>
                        </div>
                        <div className="comment-form-error-container"></div>
                        <div className="comment-form-element-container">
                            <input id="comment-form-submit" type="submit" required aria-label="comment form submit" value="Submit"/>
                        </div>
                </fieldset>
            </form>
        );
    }
    renderCommentLoginMessage(){
        return(<div id='comment-login-message'><h3>You must <Link to='/login'>log in</Link> to submit comments</h3></div>);
    }
    handleSubmit=e=>{
        e.preventDefault()
        const {title,comment}=e.target;
        // Have to add in userId when user login system is implemented.
        CommentsApiService.postOrgComment(this.props.orgId,title.value,comment.value)
            .then(res=>this.props.updateComments(this.props.orgId))
            .catch(this.state.error);
    }
    render(){
        return(
            <div className="organization-comment-form-container">
                {TokenService.hasAuthToken()?this.renderCommentForm():this.renderCommentLoginMessage()}
            </div>
        );
	}
}