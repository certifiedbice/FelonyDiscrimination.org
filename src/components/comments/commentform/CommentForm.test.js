import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import CommentForm from './CommentForm';

let orgId='5';

it('renders without crashing',()=>{
    const div=document.createElement('div');
    ReactDOM.render(<CommentForm orgId={orgId}/>,div);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<CommentForm orgId={orgId}/>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});