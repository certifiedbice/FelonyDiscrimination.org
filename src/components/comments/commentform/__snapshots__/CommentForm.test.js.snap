// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="organization-comment-form-container"
>
  <form
    aria-label="Comment form"
    id="comment-form"
    name="comment-form"
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        <h3>
          Submit a Comment
        </h3>
      </legend>
      <div
        className="comment-form-element-container"
      >
        <label>
          Title:
        </label>
        <input
          aria-labelledby="Comment form email"
          className="comment-form-element"
          id="comment-form-input"
          name="title"
          placeholder="Search"
          required={true}
          type="text"
        />
      </div>
      <div
        className="comment-form-element-container"
      >
        <label>
          Message:
        </label>
        <textarea
          aria-labelledby="review-form-message"
          className="comment-form-element"
          cols="27"
          id="comment-form-message"
          name="content"
          required={true}
          rows="6"
        />
      </div>
      <div
        className="comment-form-error-container"
      />
      <div
        className="comment-form-element-container"
      >
        <input
          aria-label="comment form submit"
          id="comment-form-submit"
          required={true}
          type="submit"
          value="Submit"
        />
      </div>
    </fieldset>
  </form>
</div>
`;
